config/string.c

```diff
@@ -186,6 +186,40 @@ static intptr_t string_native_get(const struct ConfigSet *cs, void *var,
   return (intptr_t) str;
 }
 
+/**
+ * string_string_plus_equals - Concat String to a string - Implements ConfigSetType::string_plus_equals
+ */
+static int string_string_plus_equals(const struct ConfigSet *cs, void *var, const struct ConfigDef *cdef,
+                             const char *value, struct Buffer *err)
+{
+  /* Skip if the value is missing or empty string*/
+  if (!value || (value && (value[0] == '\0')))
+    return CSR_SUCCESS | CSR_SUC_NO_CHANGE;
+
+  int rc = CSR_SUCCESS;
+
+  if (var)
+  {
+    char *str = mutt_buffer_strdup(var);
+    mutt_str_cat(str, mutt_str_len(str) + 1 + mutt_str_len(value), value);
```

Comment:
`str` is only `strlen(var)` long, but you're telling `mutt_str_cat()` that it's much longer.

Reply:
According to the documentation I understood it the same way you're telling me now. 

I tried this
```c
mutt_str_cat(str, mutt_str_len(str), value);
```

But it was shortening the original string `str` by 1 character. After adding the `1`, it stayed untouched + adding `mutt_str_len(value)` resolved it. I believe there's something wrong.

Reply:
`mutt_str_cat()` doesn't realloc the string, it expects a buffer with enough room for both strings.
`strdup(var)` is a block of memory `strlen(var)+1` long, however you're telling `mutt_str_cat()` that it's longer which will lead to a buffer overflow.
You need to measure the two strings then alloc some memory big enough for both, then copy the text.

Reply:
`mutt_str_asprintf()`s what you need (@flatcap finally remembers)
It allocs the space for you.

```c
char *new_value = NULL;
mutt_str_asprintf(&new_value, "%s%s", orig, add);
```

Reply:
this works fine. But when I `unset` the variable or set it to empty string `""`, then the original value is `"(null)"`, I have no idea where this origins.

Reply:
Hmm... I don't see that with the code, here.
Where do you see the `"(null)"` string?